common/zt_logSanitizer_test.go

```diff
@@ -0,0 +1,72 @@
+// Copyright © Microsoft <wastore@microsoft.com>
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package common
+
+import (
+	chk "gopkg.in/check.v1"
+)
+
+type logSanitizerSuite struct{}
+
+var _ = chk.Suite(&logSanitizerSuite{})
+
+func (s *logSanitizerSuite) TestLogSanitizer(c *chk.C) {
+
+	cases := []struct {
+		raw               string
+		expectedSanitized string
+	}{
+		{"string with no secrets", "string with no secrets"},
+
+		// DON'T redact these
+		{"This is the sig that I have and x=y", "This is the sig that I have and x=y"},     // sig not in a URL
+		{"http://foo/path/with/sig/in/it?x=y", "http://foo/path/with/sig/in/it?x=y"},       // match is not in the QUERY part of a URL
+		{"http://www.signature.example.com/blah", "http://www.signature.example.com/blah"}, // another with match that's not in QUERY part the URL
+		{"http://foo?signatureevent=123&x=y", "http://foo?signatureevent=123&x=y"},         // our keyword (signature) is not the END of the key-value key name
+
+		// DO redact all of the following
+		{"http://foo?sig=somevalue&x=y", "http://foo?sig=-REDACTED-&x=y"},                                         // remainder of query string is preserved
+		{"http://foo?x=y&" + SigAzure + "=somevalue", "http://foo?x=y&sig=-REDACTED-"},                            // basic case
+		{"http://foo?x=y&sig=somevalue\r\nBlah", "http://foo?x=y&sig=-REDACTED-\r\nBlah"},                         // newline after, case preserved in other text
```

Comment:
Another case of "I don't think this will fail", but it'd be best to include a case with weird capitalization on "sig" specifically.

Reply:
Noted. Will hopefuly get to it later this week.